Fix static linking of adb/adbd

Both adb and adbd use OpenSSL, which indirectly uses zlib. Since
adb/adbd also use zlib directly -lz is included in the linker flags,
but not at the right position to ensure that static linking works: to
make it possible for OpenSSL symbols to see zlib symbols, -lz must
appear after -lcrypto.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/debian/makefiles/adb.mk
===================================================================
--- a/debian/makefiles/adb.mk
+++ b/debian/makefiles/adb.mk
@@ -41,7 +41,7 @@
 CPPFLAGS+= -I$(SRCDIR)/core/adb
 CPPFLAGS+= -I$(SRCDIR)/core/include
 
-LIBS+= -lc -lpthread -lz -lcrypto
+LIBS+= -lc -lpthread -lcrypto -lz
 
 OBJS= $(SRCS:.c=.o)
 
Index: b/debian/makefiles/adbd.mk
===================================================================
--- a/debian/makefiles/adbd.mk
+++ b/debian/makefiles/adbd.mk
@@ -44,7 +44,7 @@
 CPPFLAGS+= -I$(SRCDIR)/core/adbd
 CPPFLAGS+= -I$(SRCDIR)/core/include
 
-LIBS+= -lc -lpthread -lz -lcrypto -lcrypt
+LIBS+= -lc -lpthread -lcrypto -lz -lcrypt
 
 OBJS= $(patsubst %, %.o, $(basename $(SRCS)))
 
